library/std/src/os/horizon/thread.rs

```diff
@@ -0,0 +1,88 @@
+//! Nintendo 3DS-specific extensions to primitives in the [`std::thread`] module.
+//!
+//! All 3DS models have at least two CPU cores available to spawn threads on:
+//! The application core (appcore) and the system core (syscore). The New 3DS
+//! has an additional two cores, the first of which can also run user-created
+//! threads.
+//!
+//! Threads spawned on the appcore are cooperative rather than preemptive. This
+//! means that threads must explicitly yield control to other threads (whether
+//! via synchronization primitives or explicit calls to `yield_now`) when they
+//! are not actively performing work. Failure to do so may result in control
+//! flow being stuck in an inactive thread while the other threads are powerless
+//! to continue their work.
+//!
+//! However, it is possible to spawn one fully preemptive thread on the syscore
+//! by using a service call (https://www.3dbrew.org/wiki/APT:SetApplicationCpuTimeLimit)
```

Comment:
Could you remove the link to the syscall? 

Maybe we should make this available to the user too via `pthread`?

Reply:
We could add that functionality in a separate PR. I don't think we should expose too much via std, just enough to unblock people who want to use libraries that rely on std threads. The other 3ds specific stuff can be done via ctru-rs for now.